### PR TITLE
Add kms to s3 remote state module

### PR DESCRIPTION
**Pull request by mcgirr**
Initial attempt to add server side encryption to the `s3-remote-state` module to support 

https://www.terraform.io/docs/providers/aws/r/s3_bucket.html#server_side_encryption_configuration

for https://www.terraform.io/docs/backends/types/s3.html#kms_key_id

Contributes to https://github.com/fpco/terraform-aws-foundation/issues/310

- [ ] Update the changelog
- [ ] Make sure that modules and files are documented. This can be done inside the module and files.
- [ ] Make sure that new modules directories contain a basic README.md file.
- [ ] Make sure that the module is added to [tests/main.tf](https://github.com/fpco/terraform-aws-foundation/blob/master/tests/main.tf)
- [ ] Make sure that the linting passes on CI.
- [ ] Make sure that there is an up to date example for your code:
      - For new `modules` this would entail example code for how to use the module or some explanation in the module readme.
      - For new examples please provide a README explaining how to run the example. It's also ideal to provide a basic makefile to use the example as well.
- [ ] Make sure that there is a manual CI trigger that can test the deployment.

### CODE REVIEW

File: modules/s3-remote-state/main.tf [hunk 1]
```diff
@@ -36,12 +35,26 @@ variable "force_destroy" {
   type        = bool
 }
 
+variable "kms_key_id" {
+  description = "The ARN of a KMS Key to use for encrypting the state"
+  type        = string
+}
+
```
Comment (ketzacoatl) [inline modules/s3-remote-state/main.tf]: The use of a KMS key here is a pattern worth encoding into a module like this. TY for this update.

I guess the complimentary update is a new module which defines the KMS key to be used by this module.

File: modules/s3-remote-state/main.tf [hunk 2]
```diff
@@ -72,8 +85,6 @@ data "aws_iam_policy_document" "s3-full-access" {
   statement {
     effect = "Allow"
 
-    # find an authoritative list of valid Actions for a AWS bucket policy,
-    # I haven't been able to locate one, and the two commented out are invalid
```
Comment (ketzacoatl) [inline modules/s3-remote-state/main.tf]: Can we link to the authoritative list?

Comment (ketzacoatl) [reply modules/s3-remote-state/main.tf]: oh, I see it at the top, ignore this.

File: modules/s3-remote-state/main.tf (hunk 1 above)
Comment (mcgirr) [reply modules/s3-remote-state/main.tf]: That sounds good to me. I've opened https://github.com/fpco/terraform-aws-foundation/issues/310 and included that detail in the issue.

File: modules/s3-remote-state/main.tf (hunk 2 above)
Comment (mcgirr) [reply modules/s3-remote-state/main.tf]: :slightly_smiling_face:

**Pull request merged**